src/tests/tools/iotc_test_echoserver.h

```diff
@@ -0,0 +1,124 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __IOTC_TEST_ECHOSERVER_H__
+#define __IOTC_TEST_ECHOSERVER_H__
+
+#include <netdb.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <thread>
+#include <unistd.h>
+
+namespace iotctest {
+
+const uint16_t kTimeoutSeconds = 1;
+const uint16_t kBufferSize = 1024;
+
+/**
+ * @class
+ * @brief is for testing end to end connection.
+ *
+ * This class is an API of echo server which is used to test end to end
+ * connection between client and server.
+ */
+class EchoTestServer {
+  public:
+    enum class ServerError {
+      kSuccess = 0,
+      kError = 1,
```

Comment:
?? What does kError mean? Maybe call it `kInternalError`, or `kFailedPreconditionError` ?